test/syscalls/linux/proc.cc

```diff
@@ -511,6 +513,87 @@ TEST(ProcSelfAuxv, EntryValues) {
   EXPECT_EQ(i, proc_auxv.size());
 }
 
+
+// Just open and read a part of /proc/self/mem, check that we can read an item.
+TEST(ProcSelfMem, Basic) {
+  auto memfd =
+      ASSERT_NO_ERRNO_AND_VALUE(Open("/proc/self/mem", O_RDONLY));
+  char input[] = "hello-world";
+  char output[sizeof(input)];
+  ASSERT_THAT(pread(memfd.get(), output, sizeof(output), (off_t) (input)),
+              SyscallSucceedsWithValue(sizeof(input)));
+  ASSERT_STREQ(input, output);
+}
+
+// Perform read on /proc/[pid]/mem from same task group.
+TEST(ProcSelfMem, TaskGroup) {
+  int pfd1[2] = {};
+  int pfd2[2] = {};
+
+  ASSERT_THAT(pipe(pfd1), SyscallSucceeds());
+  ASSERT_THAT(pipe(pfd2), SyscallSucceeds());
+
+  // Create child process
+  pid_t const child_pid = fork();
+  if (child_pid == 0) {
+    // Close reading end of first pipe
+    close(pfd1[0]);
+
+    // Tell parent about location of input
+    char input[] = "hello-world";
+    off_t input_location = (off_t) input;
+    int res = WriteFd(pfd1[1], &input_location, sizeof(input_location));
+    EXPECT_EQ(res, sizeof(input_location)) << "bad size written to pipe fd";
```

Comment:
It's only safe to call async-signal-safe functions in the child process after fork() (before execve()), and googletest isn't async-signal-safe. Use macros like `TEST_CHECK` and `TEST_PCHECK` instead.